File: frontend/src/types/prospect.ts
// ─── Partner classification ────────────────────────────────────────────────

export type PartnerType =
  | "hotel_riad"           // Hôtels & Riads Maroc 3*→5*
  | "hotel_luxe"           // Hôtels luxe 5* (Maroc / EAU / ES)
  | "tour_operateur"       // Tour-opérateurs Europe
  | "agence_voyage"        // Agences voyage B2B
  | "prestataire_activites"// Prestataires activités
  | "transport"            // Transport / Transferts
  | "to_golfe"             // TO Golfe (EAU / SA / QA)
  | "mice";                // Agences MICE / Incentive

export type PipelineStage =
  | "prospection"
  | "qualification"
  | "outreach"
  | "negociation"
  | "closing"
  | "activation_ota"
  | "veille"
  | "perdu";

export type OutreachLanguage = "fr" | "en" | "ar" | "es" | "de";

// ─── Scoring breakdown (spec §4) ──────────────────────────────────────────

export interface ScoreBreakdown {
  activiteDigitale: number;    // max 25 — présence web, avis, OTAs concurrentes
  coherenceMarche: number;     // max 25 — clientèle alignée Europe / Golfe
  tailleCapacite: number;      // max 20 — nb chambres / activités, volume potentiel
  contactDecideur: number;     // max 15 — email direct ou LinkedIn DG/commercial
  liberteOta: number;          // max 15 — absence d'exclusivité OTA concurrente
}

export function scoreTotal(s: ScoreBreakdown): number {
  return (
    s.activiteDigitale +
    s.coherenceMarche +
    s.tailleCapacite +
    s.contactDecideur +
    s.liberteOta
  );
}

// ─── Core prospect entity ──────────────────────────────────────────────────

export interface Prospect {
  id: string;
  nom: string;
  type: PartnerType;
  pays: string;
  ville: string;
  region?: string;
  adresseWeb: string;
  emailContact: string;
  linkedinContact?: string;
  nomContact: string;
  posteContact: string;
  nbChambres?: number;
  capaciteDescription?: string;
  presenceBooking: boolean;
  noteBooking?: number;
  presenceExpedia: boolean;
  score: ScoreBreakdown;
  stage: PipelineStage;
  commissionStandard: number;   // %
  commissionPlancher: number;   // %
  langue: OutreachLanguage;
  dateAjout: string;            // ISO YYYY-MM-DD
  dateProchainContact?: string; // ISO YYYY-MM-DD
  notes?: string;
}

// ─── Display helpers ───────────────────────────────────────────────────────

export const PARTNER_TYPE_LABELS: Record<PartnerType, string> = {
  hotel_riad: "Hôtel / Riad",
  hotel_luxe: "Hôtel Luxe 5★",
  tour_operateur: "Tour-opérateur",
  agence_voyage: "Agence Voyage",
  prestataire_activites: "Activités",
  transport: "Transport",
  to_golfe: "TO Golfe",
  mice: "MICE",
};

export const STAGE_LABELS: Record<PipelineStage, string> = {
  prospection: "Prospection",
  qualification: "Qualification",
  outreach: "Outreach",
  negociation: "Négociation",
  closing: "Closing",
  activation_ota: "Activation OTA",
  veille: "Mis en veille",
  perdu: "Perdu",
};

export const LANGUAGE_FLAGS: Record<OutreachLanguage, string> = {
  fr: "🇫🇷",
  en: "🇬🇧",
  ar: "🇲🇦",
  es: "🇪🇸",
  de: "🇩🇪",
};

export const SCORE_THRESHOLDS = {
  OUTREACH: 75,   // required to move to outreach
  HIGH: 85,       // colour: success
  MID: 75,        // colour: warning
} as const;

export function scoreColor(total: number): "success" | "warning" | "error" {
  if (total >= SCORE_THRESHOLDS.HIGH) return "success";
  if (total >= SCORE_THRESHOLDS.MID) return "warning";
  return "error";
}

File: backend/app/models/prospect.py
import uuid
from datetime import date, datetime

from sqlalchemy import Boolean, Date, DateTime, Float, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column
from sqlalchemy.types import Uuid

from app.database import Base


class Prospect(Base):
    __tablename__ = "prospects"

    id: Mapped[uuid.UUID] = mapped_column(Uuid, primary_key=True, default=uuid.uuid4)
    nom: Mapped[str] = mapped_column(String(255), nullable=False)
    type: Mapped[str] = mapped_column(String(50), nullable=False)
    pays: Mapped[str] = mapped_column(String(100), nullable=False)
    ville: Mapped[str] = mapped_column(String(100), nullable=False)
    region: Mapped[str | None] = mapped_column(String(100), nullable=True)
    adresse_web: Mapped[str] = mapped_column(String(255), unique=True, nullable=False)
    email_contact: Mapped[str] = mapped_column(String(255), nullable=False)
    linkedin_contact: Mapped[str | None] = mapped_column(String(255), nullable=True)
    nom_contact: Mapped[str] = mapped_column(String(255), nullable=False)
    poste_contact: Mapped[str] = mapped_column(String(255), nullable=False)
    nb_chambres: Mapped[int | None] = mapped_column(Integer, nullable=True)
    capacite_description: Mapped[str | None] = mapped_column(Text, nullable=True)
    presence_booking: Mapped[bool] = mapped_column(Boolean, default=False)
    note_booking: Mapped[float | None] = mapped_column(Float, nullable=True)
    presence_expedia: Mapped[bool] = mapped_column(Boolean, default=False)

    score_activite_digitale: Mapped[int] = mapped_column(Integer, default=0)
    score_coherence_marche: Mapped[int] = mapped_column(Integer, default=0)
    score_taille_capacite: Mapped[int] = mapped_column(Integer, default=0)
    score_contact_decideur: Mapped[int] = mapped_column(Integer, default=0)
    score_liberte_ota: Mapped[int] = mapped_column(Integer, default=0)
    score_total: Mapped[int] = mapped_column(Integer, default=0)

    stage: Mapped[str] = mapped_column(String(50), default="prospection")
    commission_standard: Mapped[float] = mapped_column(Float, nullable=False)
    commission_plancher: Mapped[float] = mapped_column(Float, nullable=False)
    langue: Mapped[str] = mapped_column(String(5), nullable=False)
    date_ajout: Mapped[date] = mapped_column(Date, nullable=False)
    date_prochain_contact: Mapped[date | None] = mapped_column(Date, nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)

    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)
    updated_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
